src/overlay/OverlayManagerImpl.cpp

```diff
@@ -359,8 +359,6 @@ OverlayManagerImpl::start()
 OverlayManager::AdjustedFlowControlConfig
 OverlayManagerImpl::getFlowControlBytesConfig() const
 {
-    auto const maxTxSize = mApp.getHerder().getMaxTxSize();
```

Comment:
Is there a reason for this code move? i think it was placed here originally to ensure that by the time this function is called, we've properly loaded network configs

Reply:
Yes, I moved the assert only to the branch where `maxTxSize` is actually used because otherwise this assert fires during certain tests that start `OverlayManager` without starting `Herder`. For example, [`database is purged at overlay start`](https://github.com/bboston7/stellar-core/blob/f04f0c19a18067f6af406d5c4c0fae8e72f2d804/src/overlay/test/OverlayTests.cpp#L2201).

The reason the assert fires is that I removed `FlowControl::start`, so now [`FlowControl` calls `getFlowControlBytesConfig()` in its constructor](https://github.com/bboston7/stellar-core/blob/f04f0c19a18067f6af406d5c4c0fae8e72f2d804/src/overlay/FlowControl.cpp#L39), which itself is called in the `Peer` constructor. That means tests that start `OverlayManager` without starting `Herder` first don't know the max tx size and so `getMaxTxSize` returns `0`. I think the only reason the tests didn't run into this before is that they didn't authenticate peers (or just didn't run long enough to get to that point) and so never called `FlowControl::start`.

I fixed this by explicitly setting `cfg.PEER_FLOOD_READING_CAPACITY_BYTES` and `cfg.FLOW_CONTROL_SEND_MORE_BATCH_SIZE_BYTES` in those tests and moving the check to the case where `getFlowControlBytesConfig` needs to use `maxTxSize` to automatically set flow control parameters. I'm open to other ideas though if you'd like to restore this check. Maybe there's a different default we could use in the `FlowControl` constructor that doesn't require `Herder` to be running?

Reply:
That makes sense for the tests! In prod paths, by the time we're calling this function, we need to be sure that maxTsSize is setup (regardless of whether it's used or not), so the assert also checks that we have already loaded Soroban config. Looking at the tests, `createTestApplication` starts the app by default, and there seems to be a small number of tests that don't do that. Perhaps we could force set mMaxTxSize in those tests manually and keep the assert at the top of the function? We could introduce a new function behind BUILD_TESTS flag similar to `setMaxClassicTxSize`. Wdyt?

Reply:
I restored the assert in 80c7cba06f73f1b869fb95efe35596162405af85 and added a `setMaxTxSize` function behind `BUILD_TESTS` to fix up the tests.